pinot-perf/src/main/java/com/linkedin/pinot/perf/BenchmarkFixedIntArrayOffHeapIdMap.java

```diff
@@ -0,0 +1,156 @@
+  public void setUp() {
+    Random random = new Random(System.nanoTime());
+
+    FixedIntArray[] uniqueValues = new FixedIntArray[CARDINALITY];
+    for (int i = 0; i < uniqueValues.length; i++) {
+      uniqueValues[i] = new FixedIntArray(
+          new int[]{random.nextInt(COLUMN_CARDINALITY), random.nextInt(COLUMN_CARDINALITY), random.nextInt(
+              COLUMN_CARDINALITY)});
+    }
+
+    _values = new FixedIntArray[ROW_COUNT];
+    for (int i = 0; i < _values.length; i++) {
+      _values[i] = uniqueValues[random.nextInt(CARDINALITY)];
+    }
+  }
+
+  @TearDown
+  public void tearDown()
+      throws Exception {
+  }
+
+  // Start with mid size, with overflow
+  @Benchmark
+  @BenchmarkMode(Mode.SampleTime)
+  @OutputTimeUnit(TimeUnit.MILLISECONDS)
+  public IdMap<FixedIntArray> benchmarkOffHeapWithResizeWithCache()
+      throws IOException {
+    PinotDataBufferMemoryManager memoryManager = new DirectMemoryManager("perfTest");
+
+    IdMap<FixedIntArray> idMap =
+        new FixedIntArrayOffHeapIdMap(CARDINALITY / 10, 1000, NUM_COLUMNS, memoryManager,
+            "perfTestWithCache");
+
+    for (FixedIntArray value : _values) {
+      idMap.put(value);
+    }
+
+    memoryManager.close();
+    return idMap;
+  }
+
+  // Start with max size, no cache
+  @BenchmarkMode(Mode.SampleTime)
+  @OutputTimeUnit(TimeUnit.MILLISECONDS)
+  public IdMap<FixedIntArray> benchmarkOffHeapWithReSizeWithoutCache()
+      throws IOException {
+    PinotDataBufferMemoryManager memoryManager = new DirectMemoryManager("perfTest");
+
+    IdMap<FixedIntArray> idMap =
+        new FixedIntArrayOffHeapIdMap(CARDINALITY / 10, 0, NUM_COLUMNS, memoryManager,
+            "perfTestWithCache");
+
+    for (FixedIntArray value : _values) {
+      idMap.put(value);
+    }
+
+    memoryManager.close();
+    return idMap;
+  }
+
+  @BenchmarkMode(Mode.SampleTime)
+  @OutputTimeUnit(TimeUnit.MILLISECONDS)
+  public IdMap<FixedIntArray> benchmarkOffHeapPreSizeWithCache()
+      throws IOException {
+    PinotDataBufferMemoryManager memoryManager = new DirectMemoryManager("perfTest");
+
+    IdMap<FixedIntArray> idMap =
+        new FixedIntArrayOffHeapIdMap(CARDINALITY, 1000, NUM_COLUMNS, memoryManager, "perfTestWithCache");
+
+    for (FixedIntArray value : _values) {
+      idMap.put(value);
+    }
+
+    memoryManager.close();
+    return idMap;
+  }
+
+  // Start with max size, no cache
+  @BenchmarkMode(Mode.SampleTime)
+  @OutputTimeUnit(TimeUnit.MILLISECONDS)
+  public IdMap<FixedIntArray> benchmarkOffHeapPreSizeWithoutCache()
+      throws IOException {
+    PinotDataBufferMemoryManager memoryManager = new DirectMemoryManager("perfTest");
+
+    IdMap<FixedIntArray> idMap =
+        new FixedIntArrayOffHeapIdMap(CARDINALITY, 0, NUM_COLUMNS, memoryManager, "perfTestWithCache");
+
+    for (FixedIntArray value : _values) {
+      idMap.put(value);
+    }
+
+    memoryManager.close();
+    return idMap;
+  }
+
+  public static void main(String[] args)
```

Comment:
whats the output of this?

Reply:
It was added in the description of the PR.